src/audio/dai-zephyr.c

```diff
@@ -1227,9 +1227,23 @@ static int dai_copy(struct comp_dev *dev)
 	/* get data sizes from DMA */
 	ret = dma_get_status(dd->chan->dma->z_dev, dd->chan->index, &stat);
 	if (ret < 0) {
-		dai_report_xrun(dev, 0);
+		/* DMA status can return -EPIPE and current status content if xrun occurs */
+		if (ret == -EPIPE)
+#if XRUN_RAISES_AN_ERROR
+			dai_report_xrun(dev, 0);
```

Comment:
and this will eventually call the IPC3/4 version of the notification in subsequent PRs.

Reply:
no action needed here, btw, just making an observation.

Reply:
Some { } would be nice

Reply:
Hmm. @makarukp @softwarecki @lgirdwood I wonder if we should start by just making the xrun handling here emit warnings and not call dai_report_xrun() yet. That would allow to unblock Zephyr upstream updated.

The CONFIG_NOTIFY_HOST_ON_XRUN is very misleading as no notifies are sent to host in current code. What dai_report_xrun() is to escalate the problem to pipeline level, which will try to restart the pipeline. There is also another layer of conditional logic after dai_report_xrun(). E.g. "NO_XRUN_RECOVERY" option in src/audio/pipeline/pipeline-xrun.c at the pipeline level.

So seems more time is needed to sort out how we introduce the IPC4 notification mechanisms (which are new) and how we tackle the problem that HDA link xruns happen in common use-cases and in the past we have ignored them.

We still have xrun logic at the pipeline level driven by buffering status, so it's not like we are disabling xrun handling completely.